src/storage/core.rs

```diff
@@ -303,7 +304,31 @@ impl<K: Key> Storage<K> {
         }
     }
 
-    async fn get_any_data(safe: &Safe, key: &[u8], meta: Option<&Meta>) -> Result<Vec<u8>> {
+    async fn get_data_last(safe: &Safe, key: &[u8], meta: Option<&Meta>) -> Result<Vec<u8>> {
+        let blobs = safe.blobs.read().await;
+        let blobs_stream: FuturesOrdered<_> = blobs
```

Comment:
Looks like blob has filters check inside read_any ( https://github.com/qoollo/pearl/blob/75e5c8793216138c639d95f60ad66251b2606938/src/blob/core.rs#L252 ). So you don't need another check here

Reply:
> Core idea was to check indices concurrently and then check consistently all blobs, that are not filtered out by indices

This is correct

>should we use FuturesUnordered (check asyncronously, but analyzing results according to order) or check really consistently

The result of this check should be ordered (this is critical for correctness). But this checks should be done concurrently for better perf

Reply:
@idruzhitskiy I was confused by 2 simultaneous problems, which I solve here so thanks for found issue, resolved it